src/liballoc/sync.rs

```diff
@@ -328,6 +328,40 @@ impl<T> Arc<T> {
         Self::from_inner(Box::leak(x).into())
     }
 
+    /// Constructs a new `Arc<T>` using a weak reference to itself. Attempting
+    /// to upgrade the weak reference before this function returns will result
+    /// in a `None` value.
```

Comment:
Please add at least a doc comment... better, actual unit test functions in https://github.com/rust-lang/rust/blob/0f9949c1ac70ae0f6af43968d42bc7465af2e18c/src/liballoc/sync/tests.rs.

Reply:
That *is* a doc comment. However, I've added an example and some unit tests too.

Reply:
Sorry I mean doc *test*.